compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -397,6 +397,29 @@ impl Value {
             value => Err(InterpreterError::CannotInlineMacro { value, location }),
         }
     }
+
+    /// Creates a value that holds an `Option``.
+    /// `option_type` must be a Type referencing the `Option` type.
+    pub(crate) fn option(option_type: Type, value: Option<Value>) -> Value {
+        let Type::Struct(shared_option_type, mut generics) = option_type.clone() else {
+            panic!("Expected type to be a struct");
+        };
+
+        let shared_option_type = shared_option_type.borrow();
+        assert_eq!(shared_option_type.name.0.contents, "Option");
+
+        let t = generics.pop().expect("Expected Option to have a T generic type");
```

Comment:
Can you return an error instead of panicking though?
Possibly a type mismatch error or similar. We take the return type from the callsite so I think a type error is possible if a user tries `let foo: NotAnOption = typ.as_integer();`

Reply:
Ooh, I see. I thought `return_type` was the method's return type as in the type signature.

Reply:
Looking again, it seems `return_type` comes from the function's meta, so it should always be an Option unless someone changes the builtin signature.

I'll change it not to panic, though!

Reply:
Maybe we can have a new interpreter error that's just "unknown error" or "unexpected error" for cases where we currently panic?

Reply:
We could probably get away with returning an `Option<Value>` then at the callsite where we know `return_type` is guaranteed from the function signature we can unwrap there. Then future code can make its own decision on whether it is unreachable or an error.

Reply:
I'm not sure how. If this returns an Option, at the callsite we'd have to return an error, and I don't know what error to use.